lib/instrumentation/@azure/functions.js

```diff
@@ -47,6 +48,36 @@ module.exports = function initialize(agent, azureFunctions, _moduleName, shim, {
     return
   }
 
+  // Wrap app.setup to check if streaming is enabled.
+  // `setup` is an Object property of `azureFunctions.app`
+  // and calls Azure's underlying src/setup.ts `setup` function.
+  const originalSetupDescriptor = Object.getOwnPropertyDescriptor(azureFunctions.app, 'setup')
+  const originalSetupGetter = originalSetupDescriptor?.get
+  if (originalSetupGetter) {
+    Object.defineProperty(azureFunctions.app, 'setup', {
+      enumerable: true,
+      configurable: true,
+      get: function () {
+        const originalSetup = originalSetupGetter()
+
+        return function wrappedSetup(opts) {
+          if (opts.enableHttpStream === true) {
+            _logger.debug('Azure Functions streaming is enabled, disabling http instrumentation.')
+            isStreamingEnabled = true
+
+            // Disable instrumentation for http and https, so duplicate
+            // transactions are not created from Azure Functions
+            // streaming using HTTP.
+            agent.config.instrumentation.http.enabled = false
```

Comment:
this won't work because we load the instrumentation at runtime, there's no way to disable the instrumentation post bootstrap. this should instead skip creating the wrapped transaction and decorate the active tx from the context manager created by http instrumentation